payment-management-service/payment-management-server/src/main/java/agh/fis/payment_management/service/PaymentService.java

```diff
@@ -21,38 +25,81 @@
 
+    private PaymentDto getPaymentDtoFromPaymentEntity(PaymentEntity paymentEntity){
+            PaymentDto paymentDto = new PaymentDto();
+            paymentDto.setId(paymentEntity.getId());
+            paymentDto.setPaymentProvider(paymentEntity.getPaymentProvider());
+            paymentDto.setPaymentStatus(paymentEntity.getPaymentStatus());
+            paymentDto.setDate(paymentEntity.getDate().toString());
+            paymentDto.setPrice(paymentEntity.getPrice());
+            return paymentDto;
+    }
+
     public PaymentDto createPayment(PaymentDto paymentDto){
         java.util.Date utilDate = new java.util.Date();
-
-        PaymentEntity p = new PaymentEntity(paymentDto.getPaymentProvider(), PaymentStatus.PENDING.toString(),
-                                        new Date(utilDate.getTime()), paymentDto.getPrice());
+        PaymentEntity p = new PaymentEntity(paymentDto.getPaymentProvider(), new Date(utilDate.getTime()), paymentDto.getPrice(), PaymentStatus.PENDING);
 
         paymentRepository.save(p);
 
         paymentDto.setId(p.getId());
-
-        //        REWORK:
-        paymentDto.setPaymentStatus(PaymentStatus.PENDING);
+        paymentDto.setPaymentStatus(p.getPaymentStatus());
         paymentDto.setDate(p.getDate().toString());
 
+        logger.info("New {} payment with id: {} created",paymentDto.getPaymentStatus(), paymentDto.getId());
         return paymentDto;
     }
 
     public List<PaymentDto> getAllPayments(){
-
-        List<PaymentEntity> lpe = paymentRepository.findAll();
+        List<PaymentEntity> fetchedPaymentEntities = paymentRepository.findAll();
         List<PaymentDto> paymentDtos = new ArrayList<>();
-        for (PaymentEntity pe: lpe) {
-            PaymentDto temp = new PaymentDto();
-            temp.setId(pe.getId());
-            temp.setPaymentProvider(pe.getPaymentProvider());
-            //            REWORK:
-            temp.setPaymentStatus(PaymentStatus.PENDING);
-            temp.setDate(pe.getDate().toString());
-            temp.setPrice(pe.getPrice());
-
-            paymentDtos.add(temp);
+        for (PaymentEntity paymentEntity: fetchedPaymentEntities) {
+            paymentDtos.add(getPaymentDtoFromPaymentEntity(paymentEntity));
         }
+        logger.info("Returning a list of {} Payment Entites", paymentDtos.size());
         return paymentDtos;
     }
+
+    public PaymentDto getPaymentById(Integer id){
+        Optional<PaymentEntity> paymentEntity = paymentRepository.findById(id);
+        if (paymentEntity.isPresent()){
+            logger.info("Returning Payment with id: {}", paymentEntity.get().getId());
+            return getPaymentDtoFromPaymentEntity(paymentEntity.get());
+        }
+        logger.warn("Payment with id: {} does not exist", id);
+        throw new ResponseStatusException(HttpStatus.NO_CONTENT,  "Payment with given id does not exist");
+    }
+
+    public PaymentDto deletePaymentById(Integer id){

```

Comment:
What's the reason for returning deleted element?

Reply:
what's the reason for not returning it?

Reply:
You don't send to the client additional (redundant) data about something that actually not exists.
https://pl.wikipedia.org/wiki/KISS_(regu%C5%82a)